go/apps/afropinions/handlers.py

```
@@ -0,0 +1,67 @@
+# -*- test-case-name: go.apps.afropinions.test_handlers -*-
+import base64
+from urllib import urlencode
+
+from twisted.internet.defer import inlineCallbacks
+
+from vumi.utils import http_request_full
+from vumi import log
+
+from go.vumitools.handler import EventHandler
+from go.vumitools.api import VumiUserApi
+
+
+class YoPaymentHandler(EventHandler):
+
+    def get_user_api(self, account_key):
+        return VumiUserApi(self.dispatcher.vumi_api, account_key)
+
+    def setup_handler(self):
+        self.username = self.config['username']
+        self.password = self.config['password']
+        self.url = self.config['url']
+        self.method = self.config['method']
+        self.amount = self.config['amount']
+        self.reason = self.config['reason']
+
+    def get_auth_headers(self, username, password):
+        credentials = base64.b64encode('%s:%s' % (username, password))
+        return {
+            'Authorization': 'Basic %s' % (credentials.strip(),)
+        }
+
+    @inlineCallbacks
+    def handle_event(self, event, handler_config):
+        """
+
+        Hit the Yo payment gateway when a vxpoll is completed.
+
+        Expects 'content' to be a dict with the following keys and values:
+
+        :param from_addr:
+            The address from which the message was received and which
+            should be topped up with airtime.
+
+        """
+        if not self.url:
+            log.error('No URL configured for YoPaymentHandler')
+            return
```

Should this error perhaps not be thrown during setup_handler()? Seems a pity not to fail early.